Sources/TuistGraph/Graph/GraphOutputJSON.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+import TSCBasic
+
+public struct GraphOutputJSON: Codable, Equatable {
+    public var name: String
+    public var path: String
+    public var projects: [String: ProjectOutputJSON]
+    
+    public init(name: String, path: String, projects: [String: ProjectOutputJSON]) {
+        self.name = name
+        self.path = path
+        self.projects = projects
+    }
+    
+    public static func from(_ graph: Graph, withConfig config: Config) -> GraphOutputJSON {
+        var projects = [String: ProjectOutputJSON]()
+        for (_, project) in graph.projects {
+            // Convert packages.
+            var packages = [PackageOutputJSON]()
+            for package in project.packages {
+                switch package {
+                case .remote(let url, _):
+                    packages.append(PackageOutputJSON(kind: PackageOutputJSON.PackageKind.remote, path: url))
+                case .local(let path):
+                    packages.append(PackageOutputJSON(kind: PackageOutputJSON.PackageKind.local, path: path.pathString))
+                }
+            }
+            
+            // Convert custom schemes. These are the explicitly specified schemes not the default
+            // ones generated automatically for all the targets.
+            var schemes = [SchemeOutputJSON]()
+            for scheme in project.schemes {
+                var testTargets = [String]()
+                if let testAction = scheme.testAction {
+                    for testTarget in testAction.targets {
+                        testTargets.append(testTarget.target.name)
+                    }
+                }
+                schemes.append(SchemeOutputJSON(name: scheme.name, testActionTargets: testTargets))
+            }
+            
+            // Convert targets.
+            let didDisableAutogeneratedSchemes = config.disableAutogeneratedSchemes
+            var targets = [TargetOutputJSON]()
+            for target in project.targets {
```

Comment:
If you have specific `from` methods for a target, you could do something along the lines of:
```swift
let targets = project.targets.reduce(into: []) {
   $0.append(TargetOutput.from($1))
}
```